docs: OFO doc updates (#278)
open-feature/open-feature-operator: docs/feature_flag_configuration.md

# Feature Flag Configuration

The `FeatureFlagConfiguration` version `v1alpha2` CRD defines the a CR with the following example structure:

```yaml
apiVersion: core.openfeature.dev/v1alpha2
kind: FeatureFlagConfiguration
metadata:
    name: featureflagconfiguration-sample
spec:
    flagDSpec:
        envs:
        - name: FLAGD_PORT
          value: "8080"
    featureFlagSpec: 
        "flags": {
            "foo": {
                "state": "ENABLED",
                "variants": {
                    "bar": "BAR",
                    "baz": "BAZ"
                },
                "defaultVariant": "bar"
            }

>         }

Reviewer: I recommend switching to YAML for readability.

```

Within the CRD there are 2 main objects, namely the `flagDSpec` and the `featureFlagSpec`, both offering a different set of configuration for the injected `flagd` sidecars.

> The `featureFlagSpec` is a json object representing the flag configurations themselves, the documentation for this object can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/flag_configuration.md).

Reviewer: ```suggestion
The `featureFlagSpec` is an object representing the flag configurations themselves, the documentation for this object can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/flag_configuration.md).
```

> The `flagDSpec` has one property called `envs` which contains a list of environment variables used to override the default start command values in `flagd`, the documentation for this configuration can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md).

Reviewer: I think we need to mention here or in flagd that the env vars are also available in the workload pod for simplified configuration.

Author: docs updated
